src/main/java/duke/util/Storage.java

```diff
@@ -0,0 +1,107 @@
+package duke.util;
+
+import duke.Duke;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+
+import java.util.ArrayList;
+import java.util.Scanner;
+import java.io.FileWriter;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+
+
+public class Storage {
+    private final File file;
+    private final String filePath;
+
+    public Storage(String filePath) {
+        this.file = new File(filePath);
+        this.filePath = filePath;
+
+        try {
+            if (this.file.createNewFile()) {
+                new Ui().showToUser("New save file made");
+            } else {
+                new Ui().showToUser("File already exists meow");
+            }
+        } catch (IOException e) {
+            new Ui().showToUser("Error when construction of Storage meow");
+            System.out.println(new Ui().formatMessage(e.getMessage()));
+        }
+    }
+
+    public TaskList load() throws DukeException {
+        TaskList tl = new TaskList();
+        try {
+            Scanner sc = new Scanner(this.file);
+            while (sc.hasNext()) {
+                String[] split = sc.nextLine().split("\\|");
+                ArrayList<String> encodeSplit = new ArrayList<>();
+                for (String s : split) {
+                    encodeSplit.add(s);
+                }
+                String type = encodeSplit.get(1);
+                boolean isDone;
+                if (encodeSplit.get(2).equals("X")) {
+                    isDone = true;
+                } else {
+                    isDone = false;
+                }
+                String description;
+                switch (type) {
+                    case "T":
+                        if (encodeSplit.size() != 4) {
+                            throw new DukeException("Error with Todo load");
+                        }
+                        description = encodeSplit.get(3);
+                        tl.add(new Todo(description, isDone));
+                        break;
+                    case "D":
+                        if (encodeSplit.size() != 5) {
+                            throw new DukeException("Error with Deadline load");
+                        }
+                        description = encodeSplit.get(3);
+                        String by = encodeSplit.get(4);
+                        tl.add(new Deadline(description, by, isDone));
+                    case "E":
+                        if (encodeSplit.size() != 6) {
+                            throw new DukeException("Error with Deadline load");
+                        }
+                        description = encodeSplit.get(3);
+                        String from = encodeSplit.get(4);
+                        String to = encodeSplit.get(5);
+                        tl.add(new Event(description, from, to, isDone));
+                    default:
```

Comment:
A minor issue here is that `case` and `switch` statements should not be indented (Java coding standard) so maybe you can fix this alongside other places where `switch` is used.